src/components/sidecar/get-truncated-title.test.ts

```diff
@@ -0,0 +1,45 @@
+import getTruncatedTitle from './get-truncated-title'
```

Comment:
nit: What do you think about putting tests in a `__tests__` folder? This makes for a cleaner separation between test files and source files.

Reply:
(I like seeing tests here!)

Reply:
This is a great question, and tbh i'm not sure what I think ahah 😅

On the one hand, I've used co-located `.test` files before and have liked that it's immediately clear when contributing to source whether a function has a test file, and that it kind of raises the visibility that tests should exist. Co-locating test files also seems to create a convention that there should be one test file per source file, which I also kinda like (but now that I think about it, must have its downsides as well).

On the other hand, I've never really worked in a project that uses a `__tests__` folder outside the source files so I don't have a lot to go off of (I guess I have seen these in some more recent projects, I think? But this wasn't an explicit convention in anything I've worked in yet, IIRC).  I am curious about the perceived benefit of making that cleaner separation... but given I haven't tried it, I might as well try it now -  will move this into a `__tests__` folder! 👍 💖

Reply:
Yeah I guess it's largely personal preference. I've almost always used `__tests__`, so maybe it's just a familiarity thing haha. I do like that it clearly separates the test files, but I see your point about the clear 1:1 mapping as well. Try it out and lemme know what ya think!

Reply:
@BRKalow  oh dang i just realized that for some reason I had this in my head as the `__tests__` option:

- `src/__tests__/components/breadcrumb-bar/get-breadcrumb-links.test.ts`

and realizing now that you wrote `"a"` `__tests__` folder, not `"the"` tests, folder, so maybe what i should have been picturing is:

- `src/components/breadcrumb-bar/__tests__/get-breadcrumb-links.test.ts`

Is that accurate? If so, zero objections and fully on board with that, really dig the balance between separation and co-location.

Reply:
☝️  Updated with a `__tests__` folder within `components/sidecar`, and moved the source file so now we've got:

- Source: `src/components/sidecar/utils/get-breadcrumb-links.test.ts`
- Test: `src/components/sidecar/__tests__/get-breadcrumb-links.ts`

Let me know if this seems reasonable / matches the `__tests__` approach we're going for 🙇‍♂️

Reply:
Here's an example of what I've seen a lot in my experience. I've used 4-space indents to hopefully make the hierarchy more prominent. I had a hard time seeing it myself with just 2 spaces. 😅

- all component files are co-located in the component's folder (i.e. no __tests__ folder since there's typically only one test file based on the next point)
- utils a component uses aren't co-located to the component, but in a `utils` or similarly named folder
- utils and utils-like folders (i.e. a hooks folder) have a single `__tests__` folder and don't have any other subfolders or nesting according to how the code inside them are used

```
src/
    components/
        breadcrumb-bar/
            index.tsx
            breadcrumb-bar.module.css
            breadcrumb-bar.test.ts
    hooks/
        __tests__/
            use-count.test.ts
            use-query-string.test.ts
        use-count.ts
        use-query-string.ts
    utils/
        __tests__/
            add-some-numbers.test.ts
            get-breadcrumb-links.test.ts
        add-some-numbers.ts
        get-breadcrumb-links.ts
    ...
```

That's what I had in mind, but I think this also depends on how complex a component is and if it's being packaged up for use in other projects? If a component has a few utils that are helpful or needed to consume the component, then the structure might be a little different but I think I've mostly seen that handled by having a `packages` root folder and then basically sub-project folders within there for each thing getting packaged up.

Reply:
Yeah a colocated __tests__ folder is what I had in mind here, sorry if that wasn't clear! So anywhere we might put tests, we would add a __tests__ folder and put tests/test related code there. 👍

Reply:
@ashleemboyer @BRKalow I think since I'm still getting familiar with the approach here, I might need specific feedback on the test files in this PR 😅 Are ya'll good with the setup I have as-is:

- Source: `src/components/sidecar/utils/get-truncated-title.test.ts`
- Test: `src/components/sidecar/__tests__/get-truncated-title.ts`

If not, where should I move those two files?

Reply:
Re `__tests__` or `*.test.ts`, I would advocate for one or the other, but not both.

If we're going with `__tests__` I would say put the test file at: `src/components/sidecar/utils/__tests__/get-truncated-title.test.ts`

Reply:
In this case since `get-truncated-title` can be reused beyond sidecar, I would move these files to:

- `src/utils/get-truncated-title.ts`
- `src/utils/__tests__/get-truncated-title.test.ts`

We have a `lib` folder already, is that the same as `utils` just a different name? If so, `lib` is fine with me

Reply:
@BRKalow does webpack / next know to ignore files inside of a `__tests__` folder? I thought that file `.test.` file extensions were always needed to ensure test files aren't included in a bundle?

Reply:
If they aren't imported elsewhere there should be no problem. I usually also use the `.test.ts` extension, even if in a `__tests__` folder.